products/bigquery/terraform.yaml

```diff
@@ -0,0 +1,63 @@
+# Copyright 2019 Google Inc.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+--- !ruby/object:Provider::Terraform::Config
+legacy_name: 'bigquery'
+overrides: !ruby/object:Overrides::ResourceOverrides
+  Dataset: !ruby/object:Overrides::Terraform::ResourceOverride
+    id_format: "{{project}}:{{dataset_id}}"
+    import_format: ["{{project}}:{{dataset_id}}", "{{project}}/{{dataset_id}}", "{{dataset_id}}"]
+    examples:
+      - !ruby/object:Provider::Terraform::Examples
+        name: "bigquery_dataset_basic"
+        primary_resource_id: "dataset"
+        vars:
+          dataset_id: "example_dataset"
+    properties:
+      access: !ruby/object:Overrides::Terraform::PropertyOverride
+        default_from_api: true
+      defaultTableExpirationMs: !ruby/object:Overrides::Terraform::PropertyOverride
+        validation: !ruby/object:Provider::Terraform::Validation
+          function: 'validateDefaultTableExpirationMs'
+      datasetReference: !ruby/object:Overrides::Terraform::PropertyOverride
```

Comment:
Take a look at `flatten_object`: I think that's what you'll actually want here instead of the custom code for dataset_id. It may or may not work for project since that tends to be a bit of a special case, but if it doesn't end up working it would be great to have some comments that lay out why.

Reply:
I was not aware of it, seems to work fine. Thanks for the tip.